### ISSUE TITLE
Use Operator Lifecycle Manager to Manage Contour Operator

### ISSUE THREAD

**Issue by danehans** [projectcontour/contour-operator#88]
**Please describe the problem you have**
Use [Operator Lifecycle Manager](https://github.com/operator-framework/operator-lifecycle-manager) to manage the operator.

xref Slack discussion: https://kubernetes.slack.com/archives/C8XRH2R4J/p1603824453388600

**Comment by danehans** [comment]
xref: https://github.com/projectcontour/contour-operator/issues/141
xref: https://github.com/projectcontour/contour-operator/issues/142

**Comment by youngnick** [comment]
I'd like to understand more about this - I would rather not tie the operator exclusively to a single management framework. How does OLM help solve migration and update problems?

**Comment by danehans** [comment]
@youngnick PTAL at how OLM manages [community operators](https://docs.openshift.com/container-platform/4.7/operators/understanding/olm/olm-workflow.html) for additional information. A simple example that OLM supports is- user currently runs v1.11.0 and wants to upgrade to v1.14.0. OLM will walk through each upgrade step in the chain on behalf of the user and revert if any step fails. I see OLM as an example implementation for managing the operator. We can include add'l operator management examples as they arise. Thoughts?

**Comment by youngnick** [comment]
Ah, I see. So OLM support is just about supplying the CSVs and other metadata about versions somehow?

**Comment by danehans** [comment]
@youngnick yes, this repo would include the metadata about Contour Operator in order to ensure that it can be kept running safely in a cluster and to provide information about how updates should be applied as we publish new versions of the operator. This is very similar to traditional software packaging mechanisms, i.e. RPMs. [Xref](https://github.com/operator-framework/operator-sdk-samples/tree/master/go/memcached-operator/packagemanifests) for example.
